### PR TITLE
Add EphemeralSimulator API

### PR DESCRIPTION
**Pull request by GeorgeLyon**
Introduce `chisel3.simulator.EphemeralSimulator` for ephemeral scenarios (such as scala-cli).

This PR also adds `stepUntil` to PeekPokeAPI. This is effectively just svsim's `TICK` but is generally useful (for example in the GCD module used in the tests).

#### Type of Improvement

New feature/API

#### Release Notes

- Introduce `chisel3.simulator.EphemeralSimulator` for ephemeral scenarios (such as scala-cli)

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.5.x` or `3.6.x` depending on impact, API modification or big change: `5.0.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: src/main/scala/chisel3/simulator/PeekPokeAPI.scala [hunk 1]
```diff
@@ -13,18 +14,30 @@ object PeekPokeAPI {
...
-        simulationPort.tick(
-          timestepsPerPhase = 1,
-          maxCycles = cycles,
-          inPhaseValue = 0,
-          outOfPhaseValue = 1,
-          sentinel = None
-        )
-      }
+      val simulationPort = context.simulationPorts(clock)
+      simulationPort.tick(
```
Comment (GeorgeLyon) [inline src/main/scala/chisel3/simulator/PeekPokeAPI.scala:18]: This has slightly different semantics (the initial state of the clock port is set) which I think are more correct (encountered during testing).

File: src/main/scala/chisel3/simulator/EphemeralSimulator.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package chisel3.simulator
+
+import svsim._
+import chisel3.RawModule
+
+/**
+  * `EphemeralSimulator` provides a simple API for ephemeral invocations (such as `scala-cli` scripts) to simulate Chisel modules. To keep things really simple, it only provides the peek/poke API provides enough control while hiding some of the lower-level svsim complexity.
+  * The recommended way to use `EphemeralSimulator` is simply to `import chisel3.simulator.EphemeralSimulator._` and then call `simulate(new MyChiselModule()) { module => ... }`.
+  */
```
Comment (mwachs5) [inline src/main/scala/chisel3/simulator/EphemeralSimulator.scala:9]:
```suggestion
/** Provides a simple API for ephemeral invocations (such as `scala-cli` scripts) to 
  * simulate Chisel modules. To keep things really simple, it only provides the 
  * peek/poke API provides enough control while hiding some of the lower-level svsim 
  * complexity.
  * @example
  * ```
  * import chisel3.simulator.EphemeralSimulator._
  * ...
  * simulate(new MyChiselModule()) { module => ... }
  * ```
  */
```

scaladoc nit suggestions

File: src/main/scala/chisel3/simulator/PeekPokeAPI.scala [hunk 3]
```diff
@@ -3,7 +3,8 @@ package chisel3.simulator
 import svsim._
 import chisel3._
 
-object PeekPokeAPI {
+object PeekPokeAPI extends PeekPokeAPI {}
+trait PeekPokeAPI {
```
Comment (mwachs5) [inline src/main/scala/chisel3/simulator/PeekPokeAPI.scala:7]:
```suggestion
object PeekPokeAPI extends PeekPokeAPI

trait PeekPokeAPI {
```

File: src/main/scala/chisel3/simulator/PeekPokeAPI.scala [hunk 4]
```diff
@@ -13,18 +14,30 @@ object PeekPokeAPI {
...
+        maxCycles = cycles,
+        inPhaseValue = 0,
+        outOfPhaseValue = 1,
+        sentinel = None
+      )
+    }
+
+    /**
+      * Ticks this clock up to `maxCycles`, stops early if the specified port is equal to the specified value.
+      */
```
Comment (mwachs5) [inline src/main/scala/chisel3/simulator/PeekPokeAPI.scala:29]:
```suggestion
    /** Ticks this clock up to `maxCycles`.
      *
      * Stops early if the `sentinelPort` is equal to the `sentinelValue`.
      */
```

File: src/test/scala/chiselTests/simulator/GCD.scala [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package chiselTests.simulator
+
+import chisel3._
+
+class GCD extends Module {
```
Comment (mwachs5) [inline src/test/scala/chiselTests/simulator/GCD.scala:5]:
```suggestion
/** A simple module useful for testing Chisel generation and testing */
class GCD extends Module {
```

File: src/main/scala/chisel3/simulator/EphemeralSimulator.scala [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package chisel3.simulator
+
+import svsim._
+import chisel3.RawModule
+
+/** Provides a simple API for ephemeral invocations (such as `scala-cli` scripts) to
```
[the remark this answers is not in the document]
Comment (GeorgeLyon) [reply src/main/scala/chisel3/simulator/EphemeralSimulator.scala:6]: By "Ephemeral" I basically mean you don't care about the test artifacts once the invocation completes. I'll tweak the comment.

File: src/main/scala/chisel3/simulator/EphemeralSimulator.scala [hunk 7]
```diff
@@ -0,0 +1,47 @@
+package chisel3.simulator
+
+import svsim._
+import chisel3.RawModule
+
+/** Provides a simple API for "ephemeral" invocations (where you don't care about the artifacts after the invocation completes) to
+  * simulate Chisel modules. To keep things really simple, it only provides the
```
Comment (mwachs5) [inline src/main/scala/chisel3/simulator/EphemeralSimulator.scala:7]: there is still something wrong english-wise with the sentence "To keep things really simple, ..."

File: src/main/scala/chisel3/simulator/EphemeralSimulator.scala [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  * peek/poke API provides enough control while hiding some of the lower-level svsim
```
Comment (mwachs5) [inline src/main/scala/chisel3/simulator/EphemeralSimulator.scala:8]:
```suggestion
  * peek/poke API, which provides enough control while hiding some of the lower-level svsim
```

perhaps?

File: src/main/scala/chisel3/simulator/EphemeralSimulator.scala (hunk 7 above)
Comment (GeorgeLyon) [reply src/main/scala/chisel3/simulator/EphemeralSimulator.scala:7]: Oh, maybe this got borked when I committed a suggestion

**Review by mwachs5 — APPROVED** [review]

File: src/main/scala/chisel3/simulator/EphemeralSimulator.scala (hunk 7 above)
Comment (GeorgeLyon) [reply src/main/scala/chisel3/simulator/EphemeralSimulator.scala]: Tweaked the wording

**Pull request merged**